### PR TITLE
Improve self-hosting support

### PR DESCRIPTION
**Pull request by LoganK**
The intent is to not change any existing behavior but simply add the minimal necessary features for running the server one's self. It's working great for me so far.

If you have issues, take note that I am using the newer[1] multi-stage docker to keep the resulting image to a svelte 32MiB. 

Pull request is for the dev branch even though it appears to be behind master.

[1] Newer == January 2017 ... apparently newer than what's bundled with Ubuntu by default.

### CODE REVIEW

**Comment by LoganK** [comment]
Regarding which branch to use, it's a pain because the discordgo interface changed and you can't use latest. I used dev only because it's in the contribution guidelines. I'd be glad to rebase the changes .

File: web/web.go [hunk 1]
```diff
@@ -63,6 +65,7 @@ func init() {
 	Templates = Templates.Funcs(yagtmpl.StandardFuncMap)
 
 	flag.BoolVar(&properAddresses, "pa", false, "Sets the listen addresses to 80 and 443")
+	flag.BoolVar(&externalTls, "external-tls", false, "Serve on HTTP only.")
```
[the remark this answers is not in the document]
Comment (LoganK) [reply web/web.go]: Sounds like a good suggestion to me.

Comments on dev vs. master branch?

**Comment by LoganK** [comment]
Apologies Jonas. I force-pushed with the master branch (as dev does not compile), but went back to dev because github cannot easily change the target branch. During this process, I lost your specific change.

I have renamed the flag as requested, though. I also made a slight modification to the Dockerfile to properly catch the shutdown signal.

**Comment by jonas747** [comment]
Looks good

**Pull request merged**